scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,862 @@
+            try {
+                logger.info("Closing Record Manager");
+                recordManager.close();
+                recordManager = null;
+            } catch (IOException e) {
+                // ignored
+            }
+        }
+        return true;
+    }
+
+    private Channel getChannel(String name) throws InvalidChannelException {
+        if (inMemoryChannels.containsKey(name)) {
+            return inMemoryChannels.get(name);
+        } else if (persistedChannels.containsKey(name)) {
+            return persistedChannels.get(name);
+        }
+        throw new InvalidChannelException("Channel " + name + " does not exist");
+    }
+
+    @Override
+    public boolean createChannel(String originator, TaskId taskid, String name, boolean isPersistent)
+            throws IOException {
+        try {
+            boolean alreadyExistingChannel = deleteChannel(originator, taskid, name);
+            Channel newChannel = new Channel();
+            if (isPersistent) {
+                persistedChannels.put(name, newChannel);
+                logWithContext(taskid, null, "Created new persistent channel " + QUOTE + name + QUOTE, Level.INFO);
+                recordManager.commit();
+            } else {
+                logWithContext(taskid, null, "Created new memory channel " + QUOTE + name + QUOTE, Level.INFO);
+                inMemoryChannels.put(name, newChannel);
+            }
+            return !alreadyExistingChannel;
+        } catch (InvalidChannelException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public boolean deleteChannel(String originator, TaskId taskid, String name) throws IOException {
+        try {
+            if (persistedChannels.containsKey(name)) {
+                persistedChannels.remove(name);
+                logWithContext(taskid, null, "Deleted persistent channel " + QUOTE + name + QUOTE, Level.INFO);
+                recordManager.commit();
+                return true;
+            } else if (inMemoryChannels.containsKey(name)) {
+                inMemoryChannels.remove(name);
+                logWithContext(taskid, null, "Deleted memory channel " + QUOTE + name + QUOTE, Level.INFO);
+                return true;
+            } else {
+                return false;
+            }
+        } catch (InvalidChannelException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public boolean channelExists(String originator, TaskId taskid, String name) {
+        return inMemoryChannels.containsKey(name) || persistedChannels.containsKey(name);
+    }
+
+    @Override
+    public boolean createChannelIfAbsent(String originator, TaskId taskid, String name, boolean isPersistent)
+            throws IOException {
+        if (inMemoryChannels.containsKey(name) || persistedChannels.containsKey(name)) {
+            return false;
+        } else {
+            createChannel(originator, taskid, name, isPersistent);
```

Comment:
We should probably return the return value of `createChannel` here.

Reply:
ok